src/hint_processor/builtin_hint_processor/bigint.rs

```diff
@@ -80,7 +80,7 @@ pub fn bigint_safe_div_hint(
     let p = exec_scopes.get::<BigInt>("p")?;
 
     let k = safe_div_bigint(&(res * y - x), &p)?;
-    let (value, flag) = if BigInt::zero() < k {
+    let (value, flag) = if k.is_negative() {
```

Comment:
I saw the orders inverted when doing the suggestion. This should be `k.is_positive()`.

Reply:
```suggestion
    let (value, flag) = if k.is_positive() {
```